solidity/test/LiquidityPoolV1Converter.js

```diff
@@ -554,6 +536,70 @@ contract("LiquidityPoolV1Converter", (accounts) => {
 		});
 	}
 
+	it("verifies that it gives latest answer from oracle", async () => {
+		const converter = await initConverter(true, false, 5000);
+		await converter.setConversionFee(3000);
+
+		const amount = new BN(500);
+		await reserveToken.approve(sovrynSwapNetwork.address, amount, { from: sender });
+
+		await convert([reserveToken.address, tokenAddress, reserveToken2.address], amount, MIN_RETURN, { value: 0 });
+		const ema0 = await oracle.ema0.call();
+		const ema1 = await oracle.ema1.call();
+		expect(await oracle.latestPrice.call(reserveToken.address)).to.be.bignumber.equal(ema0);
+		expect(await oracle.latestPrice.call(reserveToken2.address)).to.be.bignumber.equal(ema1);
+	});
+
+	for (let decimal0 = 1; decimal0 <= 18; decimal0++) {
+		for (let decimal1 = 1; decimal1 <= 18; decimal1++) {
+			const maxConversionFee = 5000
+
+			const setupConverter = async () => {
+				reserveToken = await ERC20Token.new("ERC Token 1", "ERC1", decimal0, 1000000000);
+				reserveToken2 = await TestNonStandardToken.new("ERC Token 2", "ERC2", decimal1, 2000000000);
+
+				token = await SmartToken.new("Token1", "TKN1", 2);
+				tokenAddress = token.address;
+
+				const converter = await createConverter(tokenAddress, contractRegistry.address, maxConversionFee);
+				oracle = await Oracle.new(converter.address, reserveToken.address);
+				oracleAddress = oracle.address;
+
+				await converter.setOracle(oracleAddress);
+
+				await converter.addReserve(reserveToken.address, 250000);
+				await converter.addReserve(reserveToken2.address, 150000);
+				await reserveToken2.transfer(converter.address, 8000);
+				await token.issue(sender, 20000);
+
+				await reserveToken.transfer(converter.address, 5000);
+
+				await token.transferOwnership(converter.address);
+				await converter.acceptTokenOwnership();
+
+				return converter;
+			};
+
+			it("verifies that oracle gives non zero EMA for all decimal ranges", async () => {
```

Comment:
non-zero is not good enough. the actual correctness of the EMA needs to be verified.

Reply:
So the EMA should be calculated with same formula we have in contracts right?